compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -463,6 +468,12 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
       nestedApplyNeedsLift ||
       !isUnliftableFun(fun) && !tree.args.isEmpty && !tree.args.forall(LiftCoverage.noLift)
 
+    private def isContextFunctionApply(fun: Tree)(using Context): Boolean =
+      fun match
+        case Select(prefix, meth) =>
+          defn.isContextFunctionType(prefix.tpe.widen) && meth == nme.apply
```

Comment:
Consider testing `meth == nme.apply` first, as it's cheaper. You can even put it instead of `meth` in the pattern.